src/ffi/server.rs

```diff
@@ -0,0 +1,236 @@
+use http::{Request, Response};
+use std::future::Future;
+use std::pin::Pin;
+use std::ptr;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+
+use libc::{c_int, c_uint, c_void, size_t};
+
+use crate::body::Body;
+use crate::common::exec::ConnStreamExec;
+use crate::server::conn::{Connection, Http};
+use crate::service::Service;
+
+use super::error::hyper_code;
+use super::http_types::{hyper_request, hyper_response};
+use super::io::hyper_io;
+use super::task::{hyper_executor, hyper_task, WeakExec};
+
+pub struct hyper_http(Http<WeakExec>);
```

Comment:
You couldn't know this, but I'm not a fan of the name that's currently `Http` in hyper. I think this C API type might want to follow the client-side naming, so maybe `hyper_serverconn_options`? And then similarly with the related functions.